lib/SIL/SILFunction.cpp

```diff
@@ -162,11 +162,15 @@ bool SILFunction::shouldOptimize() const {
 }
 
 Type SILFunction::mapTypeIntoContext(Type type) const {
+  assert((!type->hasTypeParameter() || getGenericEnvironment()) &&
+         "SILFunction should have a generic environment");
   return GenericEnvironment::mapTypeIntoContext(
       getGenericEnvironment(), type);
 }
 
 SILType SILFunction::mapTypeIntoContext(SILType type) const {
+  assert((!type.hasTypeParameter() || getGenericEnvironment()) &&
```

Comment:
This is the only place you need the assert. The static forms of mapTypeIn/OutOfContext already assert that the type is fully concrete if the given environment is null. The issue is there's no static method equivalent for the SILType version